### PR TITLE
Add more approaches guidance

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: building/tracks/approaches.md [hunk 1]
```diff
@@ -10,22 +10,31 @@ An approach should explore how an exercise can be solved a certain way.
   - Explore an idiomatic approach
   - Explore a non-idiomatic, but interesting approach
   - Contain a meta discussion (e.g. comparing the performance of approaches)
-- Include code samples
+- Start with a (full) code sample
+  - Liberally use code samples in the rest of the document
 - Feel free to dig deep into the topic
 - Link to useful resources (e.g. documentation)
+- The snippet should showcase the core of the approach
+  - A maximum of 8 lines can be used
```
Comment (ee7) [inline building/tracks/approaches.md:18]: A new genre of rules for `configlet lint`. Exciting.

File: building/tracks/approaches.md [hunk 2]
```diff
@@ -10,22 +10,31 @@ An approach should explore how an exercise can be solved a certain way.
...
+- Keep the content fairly high-level
+  - Detailed information should go into a standalone approach page
+  - Link to the standalone approach page for more information
 - Discuss how to choose between the approaches
   - What are the trade-offs?
   - Are some approaches better suited for specific use cases?
 
+## General considerations
+
+- Some language's have different ways of writing the same thing. If some of that syntax has only become available in more recent versions of the language, consider using the "old" syntax. This makes it more likely that students will understand the code.
```
Comment (ee7) [inline building/tracks/approaches.md]:
```suggestion
- Some languages have different ways of writing the same thing. If some of that syntax has only become available in more recent versions of the language, consider using the "old" syntax. This makes it more likely that students will understand the code.
```

**Review by kytrinyx — APPROVED** [review]
These are nice improvements.

File: building/tracks/approaches.md [hunk 3]
```diff
@@ -10,22 +10,31 @@ An approach should explore how an exercise can be solved a certain way.
...
+  - Detailed information should go into a standalone approach page
+  - Link to the standalone approach page for more information
 - Discuss how to choose between the approaches
   - What are the trade-offs?
   - Are some approaches better suited for specific use cases?
 
+## General considerations
+
+- Some languages have different ways of writing the same thing. If some of that syntax has only become available in more recent versions of the language, consider using the "old" syntax. This makes it more likely that students will understand the code.
+  - Alternative syntax can be shown in the approach document
```
Comment (ErikSchierboom) [inline building/tracks/approaches.md:36]:
```suggestion
  - Alternative syntax can be shown in the approach document
- If your language has a recommended style guide, follow those guidelines where possible
```

File: building/tracks/approaches.md [hunk 4]
```diff
@@ -10,22 +10,32 @@ An approach should explore how an exercise can be solved a certain way.
...
+- If your language has a recommended style guide, follow those guidelines where possible
```
Comment (ErikSchierboom) [inline building/tracks/approaches.md]:
```suggestion
- If your language has a recommended style guide, follow those guidelines (where possible)
```

**Pull request merged**